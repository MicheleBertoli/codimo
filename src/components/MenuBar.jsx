/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import GithubLink from 'components/logos/GithubLink';
import CodimoLink from 'components/logos/CodimoLink';
import { type RouteDescription } from 'routes/Routes';

const Background = styled.div`
  align-items: center;
  background: rgba(240, 240, 240, 0.9);
  box-shadow: 0 2px 2px 0 rgba(85, 95, 110, 0.4);
  display: flex;
  width: 100%;
`;
const Container = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: 980px;
`;
const LinksContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-grow: 1;
  justify-content: space-between;
`;
const LocalLinksContainer = styled.ul`
  display: flex;
  flex-direction: row;
  font-family: 'Allerta', sans-serif;
  list-style-type: none;
  padding: 0 0 0 1rem;
  & li {
    padding: 0 0.5rem 0 0;
    & a {
      color: rgba(0, 0, 0, 0.6);
      font-size: 18px;
      text-decoration: none;
    }
  }
`;
const ExternalLinksContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
  flex-grow: 1;
  justify-content: flex-end;
  & * {
    padding: 0 0 0 0.5rem;
  }
`;

type Props = {|
  routes: Array<RouteDescription>,
|};
const MenuBar = ({ routes }: Props) => (
  <Background>
    <Container>
      <CodimoLink />

      <LinksContainer>
        <LocalLinksContainer>
          {routes.map((route, key) => (
            // eslint-disable-next-line react/no-array-index-key
            <li key={key}>
              <Link to={route.path}>{route.title}</Link>
            </li>
          ))}
        </LocalLinksContainer>

        <ExternalLinksContainer>
          <GithubLink />
        </ExternalLinksContainer>
      </LinksContainer>
    </Container>
  </Background>
);

export default MenuBar;
