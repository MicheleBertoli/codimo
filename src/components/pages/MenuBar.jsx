/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import { type RouteDescription } from 'routes';

import CodimoLink from './logos/CodimoLink';
import GithubLink from './logos/GithubLink';
import HeaderLink from './links/HeaderLink';
import RawContentContainer from './RawContentContainer';

const Background = styled.div`
  background: rgba(240, 240, 240, 0.9);
  box-shadow: 0 0 4px 4px rgba(85, 95, 110, 0.4);
  height: 55px;
  position: relative;
  width: 100%;
  z-index: 1;
`;
const ContentContainer = styled(RawContentContainer)`
  display: flex;
  height: 100%;
  justify-content: space-between;
`;
const LinksContainer = styled.div`
  display: flex;
  flex-grow: 1;
  justify-content: space-between;
`;
const LocalLinksContainer = styled.ul`
  display: flex;
  margin: 0;
  padding: 0 0 0 1.5rem;
`;
const ExternalLinksContainer = styled.div`
  align-items: center;
  display: flex;
  flex-grow: 1;
  justify-content: flex-end;
  & > * {
    padding-left: 0.5rem;
  }
`;

type Props = {|
  routes: Array<RouteDescription>,
|};
const MenuBar = ({ routes }: Props) => (
  <Background>
    <ContentContainer>
      <CodimoLink />

      <LinksContainer>
        <LocalLinksContainer>
          {routes.map((route, key) => (
            // eslint-disable-next-line react/no-array-index-key
            <HeaderLink key={key} title={route.title}>
              {route.children.map((child, key2) => (
                // eslint-disable-next-line react/no-array-index-key
                <HeaderLink key={key2} to={`/${route.game}/${child.path}`} title={child.title} />
              ))}
            </HeaderLink>
          ))}
        </LocalLinksContainer>

        <ExternalLinksContainer>
          <GithubLink />
        </ExternalLinksContainer>
      </LinksContainer>
    </ContentContainer>
  </Background>
);

export default MenuBar;
