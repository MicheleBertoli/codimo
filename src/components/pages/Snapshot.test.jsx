/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 * @jest-environment node
 */
import React from 'react';
import {
  MemoryRouter as Router,
} from 'react-router-dom';
import renderer from 'react-test-renderer';

import 'test/BlocklyMock';
import 'jest-styled-components';

import CodimoLink from './logos/CodimoLink';
import FooterLink from './links/FooterLink';
import GithubLink from './logos/GithubLink';
import RawContentContainer from './RawContentContainer';
import MenuBar from './MenuBar';
import PageContainer from './PageContainer';
import Loading from './Loading';

describe('Components snapshots', () => {
  it('renders the links correctly', () => {
    const footerLinkTree = renderer.create(
      <FooterLink href="http://helloworld.test" text="Hello dear" />,
    ).toJSON();
    const codimoLinkTree = renderer.create(<Router><CodimoLink /></Router>).toJSON();
    const githubLinkTree = renderer.create(<GithubLink />).toJSON();

    expect(footerLinkTree).toMatchStyledComponentsSnapshot();
    expect(codimoLinkTree).toMatchStyledComponentsSnapshot();
    expect(githubLinkTree).toMatchStyledComponentsSnapshot();
  });
  it('renders the raw components correctly', () => {
    const rawContentContainerTree = renderer.create(<RawContentContainer />).toJSON();

    expect(rawContentContainerTree).toMatchStyledComponentsSnapshot();
  });
  it('renders the Loading correctly', () => {
    const tree = renderer.create(<Loading />).toJSON();

    expect(tree).toMatchStyledComponentsSnapshot();
  });
  it('renders the MenuBar correctly', () => {
    const tree = renderer.create(<Router><MenuBar routes={[]} /></Router>).toJSON();

    expect(tree).toMatchStyledComponentsSnapshot();
  });
  it('renders the PageContainer correctly', () => {
    const tree = renderer.create(<PageContainer />).toJSON();

    expect(tree).toMatchStyledComponentsSnapshot();
  });
});
