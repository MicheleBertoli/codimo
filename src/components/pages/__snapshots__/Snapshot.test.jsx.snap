// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshots renders the Loading correctly 1`] = `
.kiqOHi {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.iacFuA {
  color: #d3823f;
  font-size: 2em;
}

<div
  className="Loading__LoadingAnimal-eGmvca kiqOHi"
>
  <img
    src="/images/loading-animal.gif"
  />
  <div
    className="Loading__RandomText-cHlNfX iacFuA"
  >
    Armando la línea numérica
  </div>
</div>
`;

exports[`Components snapshots renders the MenuBar correctly 1`] = `
.cjTxT {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-decoration: none;
}

.cjTxT:active img {
  -webkit-animation: iVXCSc 0.245s linear infinite;
  animation: iVXCSc 0.245s linear infinite;
}

.dALwAk {
  color: #f9524e;
  font-size: 1.5rem;
  font-weight: bold;
  margin-left: 3px;
  text-shadow: 0 1px 2px rgba(85, 95, 110, 0.4);
}

.tlHEY {
  filter: drop-shadow(0 1px 2px rgba(85, 95, 110, 0.4));
}

.htBNpY {
  height: 22px;
  position: relative;
  vertical-align: middle;
}

.htBNpY path {
  fill: rgba(0, 0, 0, 0.6);
}

.liMkSd {
  margin: 0 auto;
  width: 980px;
}

.kBhrvE {
  background: rgba(240, 240, 240, 0.9);
  box-shadow: 0 0 4px 4px rgba(85, 95, 110, 0.4);
  height: 55px;
  position: relative;
  width: 100%;
  z-index: 1;
}

.eWnVBG {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.fwMiVN {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.cXTvmf {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0 0 0 1.5rem;
}

.czunCm {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.czunCm > * {
  padding-left: 0.5rem;
}

<div
  className="MenuBar__Background-fWfgRq kBhrvE"
>
  <div
    className="MenuBar__ContentContainer-hKhuIW eWnVBG RawContentContainer__RawContentContainer-cpAunS liMkSd"
  >
    <a
      className="CodimoLink__HomeLink-jiMSkB cjTxT"
      href="/"
      onClick={[Function]}
    >
      <img
        className="CodimoLink__ShadowedImg-GmRrH tlHEY"
        src="/images/logo-title.png"
      />
      <span
        className="CodimoLink__Title-hhtJeH dALwAk"
      >
        Códimo
      </span>
    </a>
    <div
      className="MenuBar__LinksContainer-jAqpOR fwMiVN"
    >
      <ul
        className="MenuBar__LocalLinksContainer-jhZUMN cXTvmf"
      />
      <div
        className="MenuBar__ExternalLinksContainer-JhLoa czunCm"
      >
        <a
          href="https://github.com/lgraziani2712/codimo"
          target="_blank"
        >
          <svg
            aria-labelledby="github-svg"
            className="ExternalLinkHoc__SvgIcon-cfbobQ htBNpY"
            role="img"
            viewBox="0 0 32.6 31.8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="github-svg"
            >
              Github
            </title>
            <path
              d="M16.3 0C7.3 0 0 7.3 0 16.3c0 7.2 4.7 13.3 11.1 15.5.8.1 1.1-.4 1.1-.8v-2.8c-4.5 1-5.5-2.2-5.5-2.2-.7-1.9-1.8-2.4-1.8-2.4-1.5-1 .1-1 .1-1 1.6.1 2.5 1.7 2.5 1.7 1.5 2.5 3.8 1.8 4.7 1.4.1-1.1.6-1.8 1-2.2-3.6-.4-7.4-1.8-7.4-8.1 0-1.8.6-3.2 1.7-4.4-.1-.3-.7-2 .2-4.2 0 0 1.4-.4 4.5 1.7 1.3-.4 2.7-.5 4.1-.5 1.4 0 2.8.2 4.1.5 3.1-2.1 4.5-1.7 4.5-1.7.9 2.2.3 3.9.2 4.3 1 1.1 1.7 2.6 1.7 4.4 0 6.3-3.8 7.6-7.4 8 .6.5 1.1 1.5 1.1 3V31c0 .4.3.9 1.1.8 6.5-2.2 11.1-8.3 11.1-15.5C32.6 7.3 25.3 0 16.3 0z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components snapshots renders the PageContainer correctly 1`] = `
.kcEwZw {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.kcEwZw:hover {
  color: #dadada;
  text-decoration: underline;
}

.liMkSd {
  margin: 0 auto;
  width: 980px;
}

.bJokCq {
  margin-bottom: 10rem;
  -webkit-user-select: initial;
  -moz-user-select: initial;
  -ms-user-select: initial;
  user-select: initial;
}

.dCzalx {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.hEPong {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #428bca;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  bottom: 0;
  height: 10rem;
  margin: 0;
  position: fixed;
  width: 100%;
  z-index: -1;
}

<div
  className="PageContainer__Wrapper-kCvKOC bJokCq"
>
  <div />
  <div
    className="PageContainer__Footer-gzRtKy hEPong"
  >
    <div
      className="PageContainer__ContentContainer-cRjYKL dCzalx RawContentContainer__RawContentContainer-cpAunS liMkSd"
    >
      <span>
        Equipo de desarrollo de Códimo. 
        <a
          className="FooterLink__Link-eeoshK kcEwZw"
          href="https://www.unrn.edu.ar/"
          target="_blank"
        >
          UNRN
        </a>
        .
      </span>
      <span>
        Distribuído bajo licencia 
        <a
          className="FooterLink__Link-eeoshK kcEwZw"
          href="https://github.com/lgraziani2712/codimo/blob/master/LICENSE"
          target="_blank"
        >
          MIT
        </a>
        .
      </span>
    </div>
  </div>
</div>
`;

exports[`Components snapshots renders the links correctly 1`] = `
.kcEwZw {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.kcEwZw:hover {
  color: #dadada;
  text-decoration: underline;
}

<a
  className="FooterLink__Link-eeoshK kcEwZw"
  href="http://helloworld.test"
  target="_blank"
>
  Hello dear
</a>
`;

exports[`Components snapshots renders the links correctly 2`] = `
.cjTxT {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-decoration: none;
}

.cjTxT:active img {
  -webkit-animation: iVXCSc 0.245s linear infinite;
  animation: iVXCSc 0.245s linear infinite;
}

.dALwAk {
  color: #f9524e;
  font-size: 1.5rem;
  font-weight: bold;
  margin-left: 3px;
  text-shadow: 0 1px 2px rgba(85, 95, 110, 0.4);
}

.tlHEY {
  filter: drop-shadow(0 1px 2px rgba(85, 95, 110, 0.4));
}

<a
  className="CodimoLink__HomeLink-jiMSkB cjTxT"
  href="/"
  onClick={[Function]}
>
  <img
    className="CodimoLink__ShadowedImg-GmRrH tlHEY"
    src="/images/logo-title.png"
  />
  <span
    className="CodimoLink__Title-hhtJeH dALwAk"
  >
    Códimo
  </span>
</a>
`;

exports[`Components snapshots renders the links correctly 3`] = `
.htBNpY {
  height: 22px;
  position: relative;
  vertical-align: middle;
}

.htBNpY path {
  fill: rgba(0, 0, 0, 0.6);
}

<a
  href="https://github.com/lgraziani2712/codimo"
  target="_blank"
>
  <svg
    aria-labelledby="github-svg"
    className="ExternalLinkHoc__SvgIcon-cfbobQ htBNpY"
    role="img"
    viewBox="0 0 32.6 31.8"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="github-svg"
    >
      Github
    </title>
    <path
      d="M16.3 0C7.3 0 0 7.3 0 16.3c0 7.2 4.7 13.3 11.1 15.5.8.1 1.1-.4 1.1-.8v-2.8c-4.5 1-5.5-2.2-5.5-2.2-.7-1.9-1.8-2.4-1.8-2.4-1.5-1 .1-1 .1-1 1.6.1 2.5 1.7 2.5 1.7 1.5 2.5 3.8 1.8 4.7 1.4.1-1.1.6-1.8 1-2.2-3.6-.4-7.4-1.8-7.4-8.1 0-1.8.6-3.2 1.7-4.4-.1-.3-.7-2 .2-4.2 0 0 1.4-.4 4.5 1.7 1.3-.4 2.7-.5 4.1-.5 1.4 0 2.8.2 4.1.5 3.1-2.1 4.5-1.7 4.5-1.7.9 2.2.3 3.9.2 4.3 1 1.1 1.7 2.6 1.7 4.4 0 6.3-3.8 7.6-7.4 8 .6.5 1.1 1.5 1.1 3V31c0 .4.3.9 1.1.8 6.5-2.2 11.1-8.3 11.1-15.5C32.6 7.3 25.3 0 16.3 0z"
    />
  </svg>
</a>
`;

exports[`Components snapshots renders the raw components correctly 1`] = `
.liMkSd {
  margin: 0 auto;
  width: 980px;
}

<div
  className="RawContentContainer__RawContentContainer-cpAunS liMkSd"
/>
`;
